Mathlib/Algebra/Order/Hom/Basic.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+
+import Mathlib.Tactic.Positivity
+import Mathlib.Data.FunLike.Basic
+
+/-!
+# Algebraic order homomorphism classes
+
+This file defines hom classes for common properties at the intersection of order theory and algebra.
+
+## Typeclasses
+
+* `NonNegHomClass`: Homs are nonnegative: `∀ f a, 0 ≤ f a`
+* `SubAdditiveHomClass`: Homs are subadditive: `∀ f a b, f (a + b) ≤ f a + f b`
+* `SubMultiplicativeHomClass`: Homs are submultiplicative: `∀ f a b, f (a * b) ≤ f a * f b`
+* `MulLEAddHomClass`: `∀ f a b, f (a * b) ≤ f a + f b`
+* `NonArchimedeanHomClass`: `∀ a b, f (a + b) ≤ max (f a) (f b)`
+
+## TODO
+
+Finitary versions of the current lemmas.
+-/
+
+open Function
+
+variable {ι F α β γ δ : Type _}
+
+/-- `NonNegHomClass F α β` states that `F` is a type of nonnegative morphisms. -/
+class NonNegHomClass (F : Type _) (α β : outParam (Type _)) [outParam (Zero β)] [outParam (LE β)]
+  extends FunLike F α fun _ => β where
+  /-- `map_nonneg` is the statement that the image of any element is non negative. -/
```

Comment:
```suggestion
  /-- the image of any element is non negative. -/
```
And etc below. No need to include the name of the field in the doc-string, and better to avoid "is that statement that" filler.